### DISCUSSION TITLE
Is there global default for up-headers?

### DISCUSSION THREAD

**Discussion by newk5** [unpoly/unpoly#325] [Q&A] [answered]
In my app, all calls to my backend contain a token that is used for authorization, this token is a big string and different per session so I can't add the up-headers attribute to all ```<a>``` tags. I know I can use up.render and specify the headers there, but what about achors?

Is there some sort of global config I can use to specify global headers that are included every time a  ```<a>``` is clicked?

**Comment by triskweline** [comment, marked as answer]
To add a header to all requests:

```js
up.on('up:request:load', function(event) {
  event.request.headers ||= {}
  event.request.headers['Session-Token'] = 'secret'
})
```

To only add a header to requests from a link:

```js
up.on('up:link:follow', function(event) {
  event.renderOptions.headers ||= {}
  event.renderOptions.headers['Session-Token'] = 'secret'
})
```

**Reply by newk5** [reply]
Thank you, that worked perfectly 👍
